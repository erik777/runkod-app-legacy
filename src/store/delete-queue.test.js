import reducer, {
  setAct,
  startAct,
  finishAct,
  okAct,
  errorAct,
  resetAct
} from "./delete-queue"

let state = undefined;

it('1- default state', () => {
  expect(reducer(state, {})).toMatchSnapshot();
});

it('2- set. 2 items in "files".', () => {
  const files = [
    {_id: 'foo', parent: '/', label: 'style.css'}, {_id: 'bar', parent: '/', label: 'main.js'}
  ];
  const act = setAct(files);
  state = reducer(state, act);
  expect(state).toMatchSnapshot();
});

it('3- start', () => {
  const act = startAct();
  state = reducer(state, act);
  expect(state).toMatchSnapshot();
});

it('5- file ok. 1 item in "files", 1 items in "completed".', () => {
  const act = okAct({_id: 'foo', parent: '/', label: 'style.css'});
  state = reducer(state, act);
  expect(state).toMatchSnapshot();
});

it('7- file error. 0 item in "files", 1 items in "completed", 1 item in "failed". ', () => {
  const act = errorAct({_id: 'bar', parent: '/', label: 'main.js'});
  state = reducer(state, act);
  expect(state).toMatchSnapshot();
});

it('8- finish', () => {
  const act = finishAct();
  state = reducer(state, act);
  expect(state).toMatchSnapshot();
});

it('9- reset', () => {
  const act = resetAct();
  state = reducer(state, act);
  expect(state).toMatchSnapshot();
});
