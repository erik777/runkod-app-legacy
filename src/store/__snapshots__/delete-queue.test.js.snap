// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1- default state 1`] = `
Object {
  "completed": Array [],
  "failed": Array [],
  "files": Array [],
  "inProgress": false,
  "log": Array [],
  "show": false,
}
`;

exports[`2- set. 2 items in "files". 1`] = `
Object {
  "completed": Array [],
  "failed": Array [],
  "files": Array [
    Object {
      "_id": "foo",
      "label": "style.css",
      "parent": "/",
    },
    Object {
      "_id": "bar",
      "label": "main.js",
      "parent": "/",
    },
  ],
  "inProgress": false,
  "log": Array [],
  "show": true,
}
`;

exports[`3- start 1`] = `
Object {
  "completed": Array [],
  "failed": Array [],
  "files": Array [
    Object {
      "_id": "foo",
      "label": "style.css",
      "parent": "/",
    },
    Object {
      "_id": "bar",
      "label": "main.js",
      "parent": "/",
    },
  ],
  "inProgress": true,
  "log": Array [],
  "show": true,
}
`;

exports[`5- file ok. 1 item in "files", 1 items in "completed". 1`] = `
Object {
  "completed": Array [
    Object {
      "_id": "foo",
      "label": "style.css",
      "parent": "/",
    },
  ],
  "failed": Array [],
  "files": Array [
    Object {
      "_id": "bar",
      "label": "main.js",
      "parent": "/",
    },
  ],
  "inProgress": true,
  "log": Array [
    Object {
      "msg": "/style.css deleted",
      "type": "success",
    },
  ],
  "show": true,
}
`;

exports[`7- file error. 0 item in "files", 1 items in "completed", 1 item in "failed".  1`] = `
Object {
  "completed": Array [
    Object {
      "_id": "foo",
      "label": "style.css",
      "parent": "/",
    },
  ],
  "failed": Array [
    Object {
      "_id": "bar",
      "label": "main.js",
      "parent": "/",
    },
  ],
  "files": Array [],
  "inProgress": true,
  "log": Array [
    Object {
      "msg": "/style.css deleted",
      "type": "success",
    },
    Object {
      "msg": "/main.js could not delete",
      "type": "error",
    },
  ],
  "show": true,
}
`;

exports[`8- finish 1`] = `
Object {
  "completed": Array [
    Object {
      "_id": "foo",
      "label": "style.css",
      "parent": "/",
    },
  ],
  "failed": Array [
    Object {
      "_id": "bar",
      "label": "main.js",
      "parent": "/",
    },
  ],
  "files": Array [],
  "inProgress": false,
  "log": Array [
    Object {
      "msg": "/style.css deleted",
      "type": "success",
    },
    Object {
      "msg": "/main.js could not delete",
      "type": "error",
    },
  ],
  "show": true,
}
`;

exports[`9- reset 1`] = `
Object {
  "completed": Array [],
  "failed": Array [],
  "files": Array [],
  "inProgress": false,
  "log": Array [],
  "show": false,
}
`;
